### ISSUE TITLE
Local links to use <<anchor,text>> format

### ISSUE THREAD

**Issue by briandominick** [Mogztter/asciidoc-googledocs-addon#21]
Right now when a hyperlink to a local anchor gets processed, it looks something like this:

```
#heading=h.cughuow97z4o[local link]
```
This seems to just be a direct copy, but I bet that ID is getting carried with the element being linked to, and we can grab that and use it to form proper local xref. We don't actually use a lot of local links, but these can be a little hard to convert since that hash string doesn't actually appear elsewhere in the converted, so it's not clear just what that local link lined up with.
```
<<anchor-to-heading-id-or-explicit-anchor,local link>>
```

**Comment by Mogztter** [comment]
http://stackoverflow.com/a/12244248

I think we can only support this feature if the document has a TOC :unamused:

**Comment by briandominick** [comment]
That's really interesting... but it's a fine workaround, right? If you have a lot of internal links, you could add a TOC before converting. I would find that useful.

**Comment by Mogztter** [comment]
> That's really interesting... but it's a fine workaround, right? If you have a lot of internal links, you could add a TOC before converting. I would find that useful.

Yes with this workaround I think it will be possible to resolve "link ids" (ie. `#heading=h.cughuow97z4o`) to their corresponding headings titles. So I may be able to create proper local xrefs.

However it's not possible to create a Table Of Contents. So basically if a TOC is *not* already present in the document we can't do nothing.

**Comment by briandominick** [comment]
So we can detect the presence of a toc, and we can detect internal
hyperlinks. If a doc has internal links but no toc, we can echo to the ui
that adding a toc will enable those links.

I'm happy to take a swipe at this one if you want. I just haven't looked at
the DOM or seen what we're working with yet, so can't tell if I'll be in
over my head wrestling with this stuff.

**Comment by Mogztter** [comment]
> I'm happy to take a swipe at this one if you want. I just haven't looked at
the DOM or seen what we're working with yet, so can't tell if I'll be in
over my head wrestling with this stuff.

Sure!
The entry point is the [document body](https://developers.google.com/apps-script/reference/document/document#getBody()) : [Code.gs#L49](https://github.com/Mogztter/asciidoc-googledocs-addon/blob/master/app/Code.gs#L49)

Then we iterate on [children](https://developers.google.com/apps-script/reference/document/body#getChild(Integer)) : [Code.gs#L50](https://github.com/Mogztter/asciidoc-googledocs-addon/blob/master/app/Code.gs#L50)

Each `child` is an `Element`: https://developers.google.com/apps-script/reference/document/element

An `Element` can be a `TableOfContents`: https://developers.google.com/apps-script/reference/document/table-of-contents

The Google Apps Scripts documentation is pretty clear and you should find all the information there. When in doubt you can use a Logger to debug: https://developers.google.com/apps-script/reference/base/logger

Let me know if you need help :wink:

**Comment by briandominick** [comment]
Very cool; thanks for the cheat sheet.
